packages/truffle-box/lib/utils/unbox.js

```diff
@@ -114,6 +114,23 @@ async function promptOverwrites(contentCollisions, logger = console) {
 
 async function copyTempIntoDestination(tmpDir, destination, options) {
   fs.ensureDirSync(destination);
+  if (!fs.emptyDirSync(destination)) {
```

Comment:
This method deletes the contents of the directory if it is not empty. Actually you need to do this check before the temporary directory is created anyways.

Reply:
This is so weird, how come you need a temp directory?

Reply:
Perhaps it is unnecessary, this is just the architecture that was handed to me on a silver platter.